ansible/playbooks/roles/preflight/tasks/main.yml

```diff
@@ -30,6 +30,38 @@
 
 - include_tasks: check-os.yml
 
+- name: Check routing configuration
+  shell: |-
+    set -o pipefail && \
+    ip route show default 0.0.0.0/0
+  register: default_routing_configuration_number
+  failed_when: default_routing_configuration_number.rc != 0
+  args:
+    executable: /bin/bash
+
+- name: Assert only one default routing configuration exists
+  assert:
+    that: default_routing_configuration_number.stdout_lines | length == 1
```

Comment:
It looks like configuration with multiple default routes is correct as long as all have unique metrics.

See this example:

```
root@repo:~# ip route show default
default via 192.168.20.1 dev eth0 proto static
default via 172.26.96.1 dev eth1 proto dhcp src 172.26.98.125 metric 100
# set metrics
root@repo:~# ifmetric eth1 0
root@repo:~# ifmetric eth0 1
root@repo:~# ip route show default
default via 172.26.96.1 dev eth1 proto dhcp src 172.26.98.125
default via 192.168.20.1 dev eth0 proto static metric 1
# if metric == 0 then there is no metric info printed
# routes are sorted by metrics - the lower value means higher priority
root@repo:~# ifmetric eth1 2
root@repo:~# ip route show default
default via 192.168.20.1 dev eth0 proto static metric 1
default via 172.26.96.1 dev eth1 proto dhcp src 172.26.98.125 metric 2
root@repo:~# ifmetric eth0 0  # restore initial value
root@repo:~# ip route del default via 172.26.96.1
root@repo:~# ip route show default
default via 192.168.20.1 dev eth0 proto static
# try to add second default route with the same metric
root@repo:~# ip route add default via 172.26.96.1 dev eth1
RTNETLINK answers: File exists
# retry with different metric
root@repo:~# ip route add default via 172.26.96.1 dev eth1 metric 10
root@repo:~# ip route show default
default via 192.168.20.1 dev eth0 proto static
default via 172.26.96.1 dev eth1 metric 10
```

Another example:
```
default via 192.168.178.1 dev eth0  metric 100 
default via 192.168.178.1 dev wlan1  metric 101 
default via 192.168.178.1 dev wlan0  metric 102 
```
source:
https://unix.stackexchange.com/questions/345862/is-it-possible-to-have-multiple-default-gateways-for-outbound-connections

Blog article: https://netbeez.net/blog/linux-set-route-priorities/

Ansible:
If there are two default routes with the same metric `ansible_default_ipv4` seems to have random value but if metric values differ, `ansible_default_ipv4` has value corresponding to the lower metric.

Summary:
We may want to raise error only if there are multiple default routes with the same metric.